src/runmodes.c

```diff
@@ -188,9 +190,9 @@ char *RunmodeGetActive(void)
  *
  * \return a string containing the current running mode
  */
-const char *RunModeGetMainMode(void)
+const char *RunModeGetMainMode(int index)
```

Comment:
RunModeGetMainMode is used to output the running mode through unix-socket, instead RunmodeGetPrimary returns the runmode id. 
So, yes we need both currently.